File: csharp-code/Inheritance/MultipleInheritance.cs
using System;
 
namespace Hello
{
    public interface IArea
    {
         int getArea(int l,int b);
    }
    
    public interface IPerimeter 
    {
         int getPerimeter(int l,int b);
    }
    
    public class Rectangle:IArea,IPerimeter
    {
        public int getArea(int l,int b)
        {
            return l*b;
        }
        
        public int getPerimeter(int l,int b)
        {
            return 2*(l+b);   
        }
    }
   
	public class Hello
	{
	    /*
	        C# Multiple Inheritance
	        
	        output
	        
	        Area 12
	        perimeter 
	    */
	    
	    
		public static void Main (string[] args)
		{
		    Rectangle rectangle=new Rectangle();
		    Console.WriteLine($"Area {rectangle.getArea(3,4)}");
		    Console.WriteLine($"perimeter {rectangle.getPerimeter(3,4)}");
		}
	}
}

File: Java-notes/Inheritance/MultipleInheritance.java
/*
    Multi Level Inheritance 

    *** output *** 
    Area of Rectangle is 12
    Perimeter of Rectange is 14
*/

interface Area{
    public int getArea();
}

interface Perimeter{
    public int getPerimeter();
}

class Rectangle implements Area,Perimeter{
    int l,b;
    public Rectangle(int l,int b)
    {
        this.l=l;
        this.b=b;
    }
    public int getArea()
    {
        return l*b;
    }
    
    public int getPerimeter()
    {
        return 2*(l+b);   
    }
}

public class MultipleInheritance{
    public static void main(String[] args)
    {
        Rectangle rectangle=new Rectangle(3,4);
        System.out.println("Area of Rectangle is "+rectangle.getArea());
        System.out.println("Perimeter of Rectange is "+rectangle.getPerimeter());
    }
}